### DISCUSSION TITLE
(Android) Allowing Screen Sleeping by Default

### DISCUSSION THREAD

**Discussion by peterlew** [part-cw/lambdanative#422] [Ideas]
Hello everyone! 

There's been a proposal to change the default behavior of LambdaNative Android apps from keeping the screen always on to allowing the screen to sleep (as specified by the device's screen settings). This is a simple change and should not affect anything other than this issue. If you want your app to keep the screen on, this can be accomplished by adding a line in ANDROID_java_oncreate. The wiki will be updated with instructions once this change has been made. 

This discussion is intended to give notice and welcome any feedback.

**Comment by peterlew** [comment]
The change would entail removing this line: https://github.com/part-cw/lambdanative/blob/master/loaders/android/bootstrap.java.in#L194

That line could then be added to ANDROID_java_oncreate for a particular app to prevent sleep when desired.

**Reply by karliwalti** [reply]
I don't see an issue with this for our apps. Maybe add some (commented) examples also to the demo apps?

**Comment by peterlew** [comment]
This change was made today. I added the code to prevent screen sleep in DemoSensors as an example - thanks, @karliwalti

An explanation can also be found under ANDROID_java_oncreate here: https://github.com/part-cw/lambdanative/wiki/Android-Substitutions
